docs/customization/api.rst

```diff
@@ -96,3 +106,432 @@ The next step is to modify the security configuration in ``config/packages/secur
+
+            return $data;
+        }
+
+        public function supportsNormalization($data, $format = null, $context = []): bool
+        {
+            if (isset($context[self::ALREADY_CALLED])) {
+                return false;
+            }
+
+            return $data instanceof ProductInterface;
+        }
+    }
+
+And now let's declare its service in config files:
+
+.. code-block:: yaml
+
+    # config/services.yaml
+    App\Serializer\ProductNormalizer:
+        tags:
+            - { name: 'serializer.normalizer', priority: 100 }
+
+.. warning::
+
+    As we can use only one Normalizer per resource we need to set priority higher then one from Sylius.
+    You can find priority value of Normalizer in ``src/Sylius/Bundle/ApiBundle/Resources/config/services/serializers.xml``
+
+Then we can add the new field:
+
+.. code-block:: php
+
+    //...
+    $data = $this->normalizer->normalize($object, $format, $context);
+
+    $data['additionalText'] = 'your custom text or logic that will be added to this field.';
+
+    return $data;
+    //...
+
+And your response should be extended with the new field:
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "additionalText": "my additional field with text",
+        //...
+    }
+
+Removing a field from a response
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that for some reason you want to remove some field from serialization.
+One of possible solution could be that you use serialization groups.
+Those will limit the fields from your resource, according to serialization groups that you will choose.
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+Let's assume that ``Product`` resource returns such a response:
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "translations": {
+            "en_US": {
+              "@id": "/api/v2/shop/product-translations/123",
+              "@type": "ProductTranslation",
+              "id": 123,
+              "name": "product name",
+              "slug": "product-name"
+        }
+    }
+
+Then let's say you want to remove ``translations``.
+
+Utilising serialization groups to remove fields might be quite tricky as Symfony combines all of the serialization files into one.
+The easiest solution to remove the field is to create a new serialization group and use it for fields you want to have and declare this group in the endpoint.
+
+First let's add the ``config/api_platform/Product.xml`` configuration file. See ``How to add an additional endpoint?`` for more information.
```

Comment:
```suggestion
First, let's add the ``config/api_platform/Product.xml`` configuration file. See :ref:`how-to-add-additional-endpoint` for more information.
```

If you add a reference to the "How to add an additional endpoint?" header, then you'd be able to link to it from here via `:ref: (see [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role)).

```
.. _how-to-add-additional-endpoint:
How to add an additional endpoint?
-----------------------------------
```